File: memtests/MemTest2.cs
using System.Threading;

public class MemTest2
{
  static int INNER = 10000;
  static int LOOPS = 1*1000*1000;
  
  public static void Main(string[] args)
  {
    System.Threading.Thread.Sleep(1000);
    
    byte[][] aa = new byte[1000][];
    for (int i = 0; i < LOOPS; i++) {
      byte[] a = new byte[INNER];
      aa[i % 1000] = a;
    }
  }
}

File: memtests/memtest2.c
#include <unistd.h>
#include <stdlib.h>
#include <string.h>

#define INNER (10000)
#define LOOPS (1*1000*1000)


int main()
{
  int i;
  char **aa, **aap;
  
  sleep(1);
  
  aa = malloc(sizeof(char *) * 1000);
  memset(aa, 0, sizeof(char *) * 1000);
  for (i = 0; i < LOOPS; i++) {
    char *a = malloc(INNER);
    aap = aa + (i % 1000);
    if (*aap) free(*aap);
    memset(a, 0, INNER);
    *aap = a;
  }
  
  return 0;
}
